diff --git a/Makefile b/Makefile
index 15c5388..66defab 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ include include/libs.mk
 
 sharedlibs: $(LIBRMN_SHARED) $(LIBVGRID_SHARED) $(LIBBURPC_SHARED)
 	# Copy extra libraries needed for runtime
-	[ -z "$(EXTRA_LIBS)" ] || cp -L $(EXTRA_LIBS) $(SHAREDLIB_DIR)
+	[ -z "$(EXTRA_LIBS)" ] || for extra in $(EXTRA_LIBS); do cp -L $$extra $(SHAREDLIB_DIR); chmod +w $(SHAREDLIB_DIR)/*; done
 
 .patched:
 	cd librmn  && patch -p1 < $(PWD)/patches/librmn.patch
diff --git a/__init__.py b/__init__.py
index 3321a16..a26b51e 100644
--- a/__init__.py
+++ b/__init__.py
@@ -51,6 +51,20 @@ def get_extra_setup_args (*path):
       self.copy_tree(srcdir,builddir,preserve_symlinks=1)
 
       check_call(['make', 'PWD='+builddir, 'BUILDDIR='+builddir, 'SHAREDLIB_DIR='+libdir, 'SHAREDLIB_SUFFIX='+sharedlib_suffix], cwd=builddir)
+
+      # Patch up macOS libraries
+      if sharedlib_suffix == 'dylib':
+        from glob import glob
+        dylibs = glob(libdir+'/*.dylib')
+        for dylib in dylibs:
+          check_call(['install_name_tool', '-id', '@rpath/'+os.path.basename(dylib), dylib])
+          check_call(['install_name_tool', '-add_rpath', '.', dylib])
+          for extra_lib in os.environ.get('EXTRA_LIBS','').split():
+            check_call(['install_name_tool', '-change', extra_lib, '@rpath/'+os.path.basename(extra_lib), dylib])
+          for dyl in dylibs:
+            check_call(['install_name_tool', '-change', dyl, '@rpath/'+os.path.basename(dyl), dylib])
+
+
   cmdclass['build'] = BuildSharedLibs
 
   # Force the impl and abi tags.
diff --git a/include/libs.mk b/include/libs.mk
index 6339151..847042a 100644
--- a/include/libs.mk
+++ b/include/libs.mk
@@ -29,9 +29,7 @@ LIBBURPC_SHARED = $(SHAREDLIB_DIR)/libburp_c_shared_$(LIBBURPC_VERSION).$(SHARED
 # Linux shared libraries need to be explicitly told to look in their current path for dependencies.
 %.so: FFLAGS := $(FFLAGS) -Wl,-rpath,'$$ORIGIN' -Wl,-z,origin
 
-# For MacOSX, try searching in current directory for libraries.
-# (Use similar behaviour to Windows DLLs).
-%.dylib: FFLAGS := $(FFLAGS) -dynamiclib -install_name @rpath/$@ -Wl,-rpath,.
+%.dylib: FFLAGS := $(FFLAGS) -dynamiclib
 
 $(LIBRMN_SHARED): $(LIBRMN_STATIC)
 	rm -f *.o
