diff --git a/DESCRIPTION b/DESCRIPTION
index 1517cc5..fe5331e 100644
--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1 +1 @@
-A collection of python modules that act as an augmented interface to librmn, vgrtid and some more functions https://wiki.cmc.ec.gc.ca/wiki/Python-RPN
\ No newline at end of file
+A collection of python modules that act as an augmented interface to librmn, vgrtid and some more functions https://github.com/meteokid/python-rpn
diff --git a/lib/rpnpy/burpc/__init__.py b/lib/rpnpy/burpc/__init__.py
index ffdc484..5dab7fb 100755
--- a/lib/rpnpy/burpc/__init__.py
+++ b/lib/rpnpy/burpc/__init__.py
@@ -85,22 +85,29 @@ def load_burpc_lib(burpc_version=None):
     """
     import os
     import ctypes as ct
+    # Load librmn shared library here, to resolve symbols when running on MacOSX.
+    from rpnpy import librmn
     ## import numpy  as np
     ## import numpy.ctypeslib as npct
 
+    # For windows, need to change the current directory to see the .dll files.
+    curdir = os.path.realpath(os.getcwd())
+    os.chdir(os.path.join(os.path.dirname(__file__),os.pardir,'_sharedlibs'))
+
     if burpc_version is None:
         BURPC_VERSION = os.getenv('RPNPY_BURPC_VERSION',
                                   BURPC_VERSION_DEFAULT).strip()
     else:
         BURPC_VERSION = burpc_version
-    burpc_libfile = 'libburp_c_shared' + BURPC_VERSION.strip() + '.so'
+    burpc_libfile = 'libburp_c_shared' + BURPC_VERSION.strip() + '.*'
 
+    localpath   = [os.path.realpath(os.getcwd())]
     pylibpath = os.getenv('PYTHONPATH', '').split(':')
     ldlibpath = os.getenv('LD_LIBRARY_PATH', '').split(':')
     eclibpath = os.getenv('EC_LD_LIBRARY_PATH', '').split()
     BURPC_LIBPATH = check_burpc_libpath(burpc_libfile)
     if not BURPC_LIBPATH:
-        for path in pylibpath + ldlibpath + eclibpath:
+        for path in localpath + pylibpath + ldlibpath + eclibpath:
             BURPC_LIBPATH = check_burpc_libpath(os.path.join(path.strip(),
                                                            burpc_libfile))
             if BURPC_LIBPATH:
@@ -117,6 +124,7 @@ def load_burpc_lib(burpc_version=None):
     except IOError:
         raise IOError('ERROR: cannot load libburp_c shared version: ' +
                       BURPC_VERSION)
+    os.chdir(curdir)
     return (BURPC_VERSION, BURPC_LIBPATH, libburpc)
 
 (BURPC_VERSION, BURPC_LIBPATH, libburpc) = load_burpc_lib()
diff --git a/lib/rpnpy/librmn/__init__.py b/lib/rpnpy/librmn/__init__.py
index 6255e40..097217f 100755
--- a/lib/rpnpy/librmn/__init__.py
+++ b/lib/rpnpy/librmn/__init__.py
@@ -75,19 +75,24 @@ def loadRMNlib(rmn_version=None):
     ## import numpy  as np
     ## import numpy.ctypeslib as npct
 
+    # For windows, need to change the current directory to see the .dll files.
+    curdir = os.path.realpath(os.getcwd())
+    os.chdir(os.path.join(os.path.dirname(__file__),os.pardir,'_sharedlibs'))
+
     if rmn_version is None:
         RMN_VERSION = os.getenv('RPNPY_RMN_VERSION',
                                 RMN_VERSION_DEFAULT).strip()
     else:
         RMN_VERSION = rmn_version
-    rmn_libfile = 'librmnshared' + RMN_VERSION.strip() + '.so'
+    rmn_libfile = 'librmnshared' + RMN_VERSION.strip() + '.*'
 
+    localpath   = [os.path.realpath(os.getcwd())]
     pylibpath   = os.getenv('PYTHONPATH','').split(':')
     ldlibpath   = os.getenv('LD_LIBRARY_PATH','').split(':')
     eclibpath   = os.getenv('EC_LD_LIBRARY_PATH','').split()
     RMN_LIBPATH = checkRMNlibPath(rmn_libfile)
     if not RMN_LIBPATH:
-        for path in pylibpath + ldlibpath + eclibpath:
+        for path in localpath + pylibpath + ldlibpath + eclibpath:
             RMN_LIBPATH = checkRMNlibPath(os.path.join(path.strip(), rmn_libfile))
             if RMN_LIBPATH:
                 break
@@ -95,6 +100,7 @@ def loadRMNlib(rmn_version=None):
     if not RMN_LIBPATH:
         raise IOError(-1, 'Failed to find librmn.so: ', rmn_libfile)
 
+    RMN_LIBPATH = os.path.abspath(RMN_LIBPATH)
     librmn = None
     try:
         librmn = ct.cdll.LoadLibrary(RMN_LIBPATH)
@@ -102,6 +108,7 @@ def loadRMNlib(rmn_version=None):
     except IOError as e:
         raise IOError('ERROR: cannot load librmn shared version: ' +
                       RMN_VERSION, e)
+    os.chdir(curdir)
     return (RMN_VERSION, RMN_LIBPATH, librmn)
 
 (RMN_VERSION, RMN_LIBPATH, librmn) = loadRMNlib()
diff --git a/lib/rpnpy/librmn/burp.py b/lib/rpnpy/librmn/burp.py
index 25f8cbb..6f8331e 100644
--- a/lib/rpnpy/librmn/burp.py
+++ b/lib/rpnpy/librmn/burp.py
@@ -1617,7 +1617,7 @@ def _mrbcvt_dict_full_init():
         mypath = os.path.join(AFSISIO.strip(), 'datafiles/constants',
                               _rbc.BURP_TABLE_B_FILENAME)
         if not (AFSISIO and os.path.isfile(mypath)):
-            AFSISIO2 = os.getenv('rpnpy', '/')
+            AFSISIO2 = os.getenv('rpnpy', os.path.dirname(__file__))
             mypath = os.path.join(AFSISIO2.strip(), 'share',
                                   _rbc.BURP_TABLE_B_FILENAME)
         try:
diff --git a/lib/rpnpy/vgd/__init__.py b/lib/rpnpy/vgd/__init__.py
index 981041f..803e618 100755
--- a/lib/rpnpy/vgd/__init__.py
+++ b/lib/rpnpy/vgd/__init__.py
@@ -75,22 +75,29 @@ def loadVGDlib(vgd_version=None):
     """
     import os
     import ctypes as ct
+    # Load librmn shared library here, to resolve symbols when running on MacOSX.
+    from rpnpy import librmn
     ## import numpy  as np
     ## import numpy.ctypeslib as npct
 
+    # For windows, need to change the current directory to see the .dll files.
+    curdir = os.path.realpath(os.getcwd())
+    os.chdir(os.path.join(os.path.dirname(__file__),os.pardir,'_sharedlibs'))
+
     if vgd_version is None:
         VGD_VERSION = os.getenv('RPNPY_VGD_VERSION',
                                 VGD_VERSION_DEFAULT).strip()
     else:
         VGD_VERSION = vgd_version
-    vgd_libfile = 'libdescripshared' + VGD_VERSION.strip() + '.so'
+    vgd_libfile = 'libdescripshared' + VGD_VERSION.strip() + '.*'
 
+    localpath   = [os.path.realpath(os.getcwd())]
     pylibpath   = os.getenv('PYTHONPATH','').split(':')
     ldlibpath   = os.getenv('LD_LIBRARY_PATH','').split(':')
     eclibpath   = os.getenv('EC_LD_LIBRARY_PATH','').split()
     VGD_LIBPATH = checkVGDlibPath(vgd_libfile)
     if not VGD_LIBPATH:
-        for path in pylibpath + ldlibpath + eclibpath:
+        for path in localpath + pylibpath + ldlibpath + eclibpath:
             VGD_LIBPATH = checkVGDlibPath(os.path.join(path.strip(), vgd_libfile))
             if VGD_LIBPATH:
                 break
@@ -106,6 +113,8 @@ def loadVGDlib(vgd_version=None):
     except IOError:
         raise IOError('ERROR: cannot load libdescrip shared version: ' +
                       VGD_VERSION)
+
+    os.chdir(curdir)
     return (VGD_VERSION, VGD_LIBPATH, libvgd)
 
 (VGD_VERSION, VGD_LIBPATH, libvgd) = loadVGDlib()
