diff --git a/lib/rpnpy/__init__.py b/lib/rpnpy/__init__.py
index 50ec402..684b8d4 100644
--- a/lib/rpnpy/__init__.py
+++ b/lib/rpnpy/__init__.py
@@ -7,7 +7,7 @@ if sys.version_info < (3,):
 else:
     integer_types = (int,)
     long = int
-    # xrange = range
+    range = range
 
 C_WCHAR2CHAR = lambda x: bytes(str(x).encode('ascii'))
 C_WCHAR2CHAR.__doc__ = 'Convert str to bytes'
diff --git a/setup.py b/setup.py
index 63fa648..b964a11 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,6 @@
 from setuptools import setup, Distribution, find_packages
 from distutils.command.build import build
+from wheel.bdist_wheel import bdist_wheel
 import sys
 from glob import glob
 import os
@@ -65,6 +66,11 @@ class BuildSharedLibs(build):
 
     check_call(['make', 'BUILDDIR='+builddir, 'SHAREDLIB_DIR='+sharedlib_dir, 'SHAREDLIB_SUFFIX='+sharedlib_suffix], cwd=builddir)
 
+# Force the impl and abi tags.
+class ForcedTag(bdist_wheel):
+  def get_tag(self):
+    plat_name = self.plat_name
+    return ('py2.py3','none',plat_name)
 
 setup (
   name = 'eccc_rpnpy',
@@ -84,5 +90,5 @@ setup (
     'rpnpy._sharedlibs': ['*.so','*.so.*','*.dll','*.dylib'],
   },
   distclass = BinaryDistribution if build_shared_libs else None,
-  cmdclass={'build': BuildSharedLibs} if build_shared_libs else {},
+  cmdclass={'build': BuildSharedLibs, 'bdist_wheel': ForcedTag} if build_shared_libs else {},
 )
