diff --git a/lib/rpnpy/librmn/__init__.py b/lib/rpnpy/librmn/__init__.py
index 78573a7..958744d 100755
--- a/lib/rpnpy/librmn/__init__.py
+++ b/lib/rpnpy/librmn/__init__.py
@@ -86,12 +86,13 @@ def loadRMNlib(rmn_version=None):
         RMN_VERSION = rmn_version
     rmn_libfile = 'librmnshared' + RMN_VERSION.strip() + '.so'
 
+    localpath   = [os.path.dirname(__file__)]
     pylibpath   = os.getenv('PYTHONPATH','').split(':')
     ldlibpath   = os.getenv('LD_LIBRARY_PATH','').split(':')
     eclibpath   = os.getenv('EC_LD_LIBRARY_PATH','').split()
     RMN_LIBPATH = checkRMNlibPath(rmn_libfile)
     if not RMN_LIBPATH:
-        for path in pylibpath + ldlibpath + eclibpath:
+        for path in localpath + pylibpath + ldlibpath + eclibpath:
             RMN_LIBPATH = checkRMNlibPath(os.path.join(path.strip(), rmn_libfile))
             if RMN_LIBPATH:
                 break
diff --git a/lib/rpnpy/vgd/__init__.py b/lib/rpnpy/vgd/__init__.py
index 207619d..03f362a 100755
--- a/lib/rpnpy/vgd/__init__.py
+++ b/lib/rpnpy/vgd/__init__.py
@@ -88,12 +88,13 @@ def loadVGDlib(vgd_version=None):
         VGD_VERSION = vgd_version
     vgd_libfile = 'libdescripshared' + VGD_VERSION.strip() + '.so'
 
+    localpath   = [os.path.dirname(__file__)]
     pylibpath   = os.getenv('PYTHONPATH','').split(':')
     ldlibpath   = os.getenv('LD_LIBRARY_PATH','').split(':')
     eclibpath   = os.getenv('EC_LD_LIBRARY_PATH','').split()
     VGD_LIBPATH = checkVGDlibPath(vgd_libfile)
     if not VGD_LIBPATH:
-        for path in pylibpath + ldlibpath + eclibpath:
+        for path in localpath + pylibpath + ldlibpath + eclibpath:
             VGD_LIBPATH = checkVGDlibPath(os.path.join(path.strip(), vgd_libfile))
             if VGD_LIBPATH:
                 break
