diff --git a/src/Makefile b/src/Makefile
index c1ace57..c1cb2a4 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,22 +1,12 @@
 
-ifeq ($(BASE_ARCH),$(EC_ARCH))
-   $(info ERROR: please setup the compilation environment by typing the folowing:)
-   $(info . setup.dot or . setup_gpsc.dot)
-   $(error )
-endif
-
 default: libdescrip.a
 
 dependencies.mk: *.c *.F90 *.h *.hc
 	ls *.hc *.c *.F90 | ./s.dependencies.pl > dependencies.mk ;\
 
 include dependencies.mk
-
-%.o: %.F90
-	s.f90 -c $(MY_FFLAGS) $<
-
-%.o: %.c
-	s.cc -c $(MY_CFLAGS) $<
+.SUFFIXES : .c .F90 .o
+include $(RPN_TEMPLATE_LIBS)/include/makefile_suffix_rules.inc
 
 libdescrip.a: $(OBJECTS) dependencies.mk
 	rm -f libdescrip.a && ar scru libdescrip.a *.o ;\
